src/core/Group.cpp

```diff
@@ -946,6 +973,23 @@ void Group::resolveConflict(Entry* existingEntry, Entry* otherEntry)
     }
 }
 
+void Group::resolveGroupConflict(Group* existingGroup, Group* otherGroup)
+{
+    const QDateTime timeExisting = existingGroup->timeInfo().lastModificationTime();
+    const QDateTime timeOther = otherGroup->timeInfo().lastModificationTime();
+
+    // only if the other group newer, update the existing one.
+    if (timeExisting < timeOther) {
+        qDebug("Updating group %s.", qPrintable(existingGroup->name()));
+        existingGroup->setName(otherGroup->name());
+        existingGroup->setNotes(otherGroup->notes());
+        existingGroup->setIcon(otherGroup->iconNumber());
+        existingGroup->setIcon(otherGroup->iconUuid());
```

Comment:
This will either reset the previously set iconNumber or trigger an assertion for !uuid.isNull(). You need to check which one is set: icon number or UUID.

Reply:
@phoerious you're right, the tests we're indeed failing when building in debug mode. Which makes me think, should we not build in debug mode in travis?

Reply:
We are building both in Debug and Release mode. I have no idea why the tests passed.